lib/SILOptimizer/Mandatory/DiagnoseStaticExclusivity.cpp

```diff
@@ -965,22 +965,33 @@ static PartialApplyInst *lookThroughForPartialApply(SILValue V) {
 /// recognize when the apply takes a block argument that has been initialized to
 /// a non-escaping closure.
 static void checkForViolationsInNoEscapeClosures(
-    const StorageMap &Accesses, FullApplySite FAS, AccessSummaryAnalysis *ASA,
+    const StorageMap &Accesses, ApplySite AS, AccessSummaryAnalysis *ASA,
     llvm::SmallVectorImpl<ConflictingAccess> &ConflictingAccesses) {
-
-  SILFunction *Callee = FAS.getCalleeFunction();
-  if (Callee && !Callee->empty()) {
+  SILFunction *Callee = AS.getCalleeFunction();
+  if (isa<FullApplySite>(AS) && Callee && !Callee->empty()) {
     // Check for violation with directly called closure
-    checkForViolationWithCall(Accesses, Callee, 0, FAS.getArguments(), ASA,
+    checkForViolationWithCall(Accesses, Callee, 0, AS.getArguments(), ASA,
```

Comment:
Is '0' still a valid startIndex?

Reply:
It is because that site is only used for full applies of directly called closures (it is not the recursive base case). I've reorganized the code to remove this logic from the recursive call to try to make this more clear.